@here/olp-sdk-dataservice-write/lib/client/PublishSinglePartitionRequest.ts

```diff
@@ -0,0 +1,148 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import { PublishPartition } from "@here/olp-sdk-dataservice-api/lib/publish-api-v2";
+
+/**
+ * @brief PublishSinglePartitionRequest is used to upload single partition.
+ */
+export class PublishSinglePartitionRequest {
+    private publicationId?: string;
+    private billingTag?: string;
+    private layerId?: string;
+    private data?: Blob | Buffer;
+    private partitionId?: string;
+    private metaData?: PublishPartition;
+
+    /**
+     * @brief set the ID of the publication to upload.
+     * @param id the ID of the publication to upload.
+     * @note Required.
+     * @returns reference to this object
+     */
+    public withPublicationId(id: string): PublishSinglePartitionRequest {
+        this.publicationId = id;
+        return this;
+    }
+
+    /**
+     * @brief get the ID of the publication to upload.
+     * @returns The ID of the publication to upload, required.
+     */
+    public getPublicationId(): string | undefined {
+        return this.publicationId;
+    }
+
+    /**
+     * @brief set the ID of the layer.
+     * @param id the ID of the layer.
+     * @note Required.
+     * @returns reference to this object
+     */
+    public withLayerId(id: string): PublishSinglePartitionRequest {
+        this.layerId = id;
+        return this;
+    }
+
+    /**
+     * @brief get the ID of the layer.
+     * @returns The ID of the layer, required.
+     */
+    public getLayerId(): string | undefined {
+        return this.layerId;
+    }
+
+    /**
+     * Optionally use embedded data in PublishPartition or uploadBlob() API
+     * @param data The content published directly in the metadata and encoded in base64.
+     * The size of the content is limited.
+     * @note Optional. This will be use in stream layers and present only if the message size is less than or equal to 1 MB.
+     * @returns reference to this object
+     */
+    public withData(data: Blob | Buffer): PublishSinglePartitionRequest {
+        this.data = data;
+        return this;
+    }
+
+    /**
+     * @return data previously set or undefined.
+     */
+    public getData(): Blob | Buffer | undefined {
+        return this.data;
+    }
+
+    /**
+     * @brief set the ID of the partition to upload.
+     * @param id the ID of the partition to upload.
+     * @note Required.
+     * @returns reference to this object
+     */
+    public withPartitionId(id: string): PublishSinglePartitionRequest {
+        this.partitionId = id;
+        return this;
+    }
+
+    /**
+     * @brief get the ID of the partition to upload.
+     * @returns The ID of the partition to upload.
+     */
+    public getPartitionId(): string | undefined {
+        return this.partitionId;
+    }
```

Comment:
I think this is redundant, as we also have it in PublishPartition. Please remove. Ill update the design.

Reply:
yes. removed.